exchanges/okgroup/okgroup_websocket.go

```diff
@@ -582,30 +722,112 @@ func (o *OKGroup) CalculateUpdateOrderbookChecksum(orderbookData *orderbook.Base
 	return int32(crc32.ChecksumIEEE([]byte(checksum)))
 }
 
-// GenerateDefaultSubscriptions Adds default subscriptions to websocket to be handled by ManageSubscriptions()
+// GenerateDefaultSubscriptions Adds default subscriptions to websocket to be
+// handled by ManageSubscriptions()
 func (o *OKGroup) GenerateDefaultSubscriptions() {
-	enabledCurrencies := o.GetEnabledPairs(asset.Spot)
 	var subscriptions []wshandler.WebsocketChannelSubscription
-	if o.GetAuthenticatedAPISupport(exchange.WebsocketAuthentication) {
-		defaultSubscribedChannels = append(defaultSubscribedChannels, okGroupWsSpotMarginAccount, okGroupWsSpotAccount, okGroupWsSpotOrder)
-	}
-	for i := range defaultSubscribedChannels {
-		for j := range enabledCurrencies {
-			enabledCurrencies[j].Delimiter = "-"
-			subscriptions = append(subscriptions, wshandler.WebsocketChannelSubscription{
-				Channel:  defaultSubscribedChannels[i],
-				Currency: enabledCurrencies[j],
-			})
+	assets := o.GetAssetTypes()
+	for x := range assets {
+		enabledCurrencies := o.GetEnabledPairs(assets[x])
+		if len(enabledCurrencies) == 0 {
+			continue
+		}
+
+		switch assets[x] {
+		case asset.Spot:
+			for _, c := range enabledCurrencies {
+				for y := range defaultSpotSubscribedChannels {
+					subscriptions = append(subscriptions,
+						wshandler.WebsocketChannelSubscription{
+							Channel:  defaultSpotSubscribedChannels[y],
+							Currency: o.FormatExchangeCurrency(c, asset.Spot),
```

Comment:
`GetEnabledPairs` already formats currencies

Reply:
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel futures/depth:BTC-USD_191025 doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel futures/candle300s:BTC-USD_191025 doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel futures/ticker:BTC-USD_191025 doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel futures/trade:BTC-USD_191025 doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel swap/depth:BTC-USD_SWAP doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel swap/candle300s:BTC-USD_SWAP doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel swap/ticker:BTC-USD_SWAP doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel swap/trade:BTC-USD_SWAP doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel swap/funding_rate:BTC-USD_SWAP doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel swap/mark_price:BTC-USD_SWAP doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel index/candle300s:BTCUSD doesn't exist 
[ERROR] | 25/10/2019 12:21:26 | routines.go exchange OKEX websocket error - OKEX error - 30040 message: Channel index/ticker:BTCUSD doesn't exist

Reply:
nar

Reply:
It does format them, but it gets the config format. I'll resolve this, but I'm not a fan of retrieving currencies, formatting them and formatting them again